tasks.go

```diff
@@ -144,3 +152,34 @@ func (t *Task) Wait(interval, max time.Duration) error {
 		time.Sleep(interval)
 	}
 }
+
+func (t *Task) WaitForCompleteStatus(timesNum int, steps ...int) (status bool, completed bool, err error) {
+	step := 1
+	if len(steps) > 0 && steps[0] > 1 {
+		step = steps[0]
+	}
+	timeout := time.After(time.Duration(step*timesNum) * time.Second)
+
+	for {
+		select {
+		case <-timeout:
+			return
+		default:
+			err = t.Ping()
+			if nil != err {
+				t.client.log.Debugf("task %s ping error %+v", t.UPID, err)
+				break
+			}
+			completed = "stopped" == t.Status
```

Comment:
can you add this logic to a Task.Completed or something so it can be exported and reused by implementors? if youre motivated you could do a Task.Running func too

Reply:
type Task add IsCompleted,IsRunning,IsFailed,IsSuccessful
and updated func (t *Task) Ping()